pulpcore/cli/ansible/remote.py

```diff
@@ -40,151 +57,76 @@ def remote(ctx: click.Context, pulp_ctx: PulpContext, remote_type: str) -> None:
-    requirements: Optional[str],
-    auth_url: Optional[str],
-    token: Optional[str],
-) -> EntityDefinition:
-
-    body: EntityDefinition = dict()
-    if isinstance(remote_ctx, PulpAnsibleRoleRemoteContext) and any(
-        [requirements_file, requirements, auth_url, token]
-    ):
-        raise click.ClickException("Options not valid for Role remote, see --help")
-    if requirements_file is not None:
-        body["requirements_file"] = f"{yaml.safe_load(requirements_file)}"
-    elif requirements is not None:
-        body["requirements_file"] = yaml.safe_load(f'"{requirements}"')
-    if auth_url is not None:
-        body["auth_url"] = auth_url
-    if token is not None:
-        body["token"] = token
-    return body
-
-
-remote.add_command(list_entities)
-remote.add_command(show_by_name)
-remote.add_command(destroy_by_name)
-
-
-@remote.command(name="create")
-@click.option("--name", required=True)
-@click.option("--url", required=True)
-@click.option(
-    "--requirements-file",
-    type=click.File(),
-    help=_("Collections only: a Collection requirements yaml"),
+lookup_options = [href_option, name_option]
+remote_options = [
+    click.option("--ca-cert", help=_("a PEM encoded CA certificate")),
+    click.option("--client-cert", help=_("a PEM encoded client certificate")),
+    click.option("--client-key", help=_("a PEM encode private key")),
+    click.option("--tls-validation", help=_("if True, TLS peer validation must be performed")),
+    click.option("--proxy-url", help=_("format: scheme//user:password@host:port")),
+    click.option("--username", help=_("used for authentication when syncing")),
+    click.option("--password"),
+    click.option(
+        "--download-concurrency", type=int, help=_("total number of simultaneous connections")
+    ),
+    click.option("--policy", help=_("policy to use when downloading")),
+    click.option(
+        "--total-timeout",
+        type=float,
+        help=_("aiohttp.ClientTimeout.total for download connections"),
+    ),
+    click.option(
+        "--connect-timeout",
+        type=float,
+        help=_("aiohttp.ClientTimeout.connect for download connections"),
+    ),
+    click.option(
+        "--sock-connect-timeout",
+        type=float,
+        help=_("aiohttp.ClientTimeout.sock_connect for download connections"),
+    ),
+    click.option(
+        "--sock-read-timeout",
+        type=float,
+        help=_("aiohttp.ClientTimeout.sock_read for download connections"),
+    ),
+    click.option("--rate-limit", type=int, help=_("limit download rate in requests per second")),
+]
+collection_options = [
+    click.option(
+        "--requirements-file",
+        callback=_requirements_callback,
+        type=click.File(),
+        help=_("Collections only: a Collection requirements yaml"),
+    ),
+    click.option(
+        "--requirements",
+        callback=_requirements_callback,
+        help=_("Collections only: a string of a requirements yaml"),
+    ),
```

Comment:
If i read the context class correctly, this option is supposed to result in `"requirements_file"` as well, right?
I wonder what happens if you have two options filling the same variable:

```suggestion
    click.option(
        "--requirements",
        "requirements_file",
        callback=_requirements_callback,
        help=_("Collections only: a string of a requirements yaml"),
    ),
```

I'd be ok with a last one wins approach. ;)

Reply:
Yeah this didn't work.